app/models/concerns/episode_media.rb

```diff
@@ -0,0 +1,118 @@
+require "active_support/concern"
+
+module EpisodeMedia
```

Comment:
These methods pretty much all existed in `Episode`.  Moved/grouped/sorted/standardized them here:

- Try to make them start with `media_` or `complete_media`
- More consistency around checking content status/medium
- Better validation on segment_counts